spiketoolkit/postprocessing/postprocessing_tools.py

```diff
@@ -829,41 +847,129 @@ def compute_channel_spiking_activity(recording, channel_ids=None, detect_thresho
+            peak: str
+                If maximum channel has to be found among negative peaks ('neg'), positive ('pos') or
+                both ('both' - default)
+            frames_before: int
+                Frames before peak to compute amplitude
+            frames_after: float
+                Frames after peak to compute amplitude
+            max_spikes_per_unit: int
+                The maximum number of spikes to extract per unit
+            memmap: bool
+                If True, waveforms are saved as memmap object (recommended for long recordings with many channels)
+            seed: int
+                Random seed for extracting random waveforms
+            save_property_or_features: bool
+                If True (default), waveforms are saved as features of the sorting extractor object
+            recompute_info: bool
+                If True, waveforms are recomputed (default False)
+            n_jobs: int
+                Number of jobs for parallelization. Default is None (no parallelization).
+            joblib_backend: str
+                The backend for joblib. Default is 'loky'.
+            verbose: bool
+                If True output is verbose
+
+    Returns
+    -------
+    coms: list
+        List of int containing extracted COMs for each unit
+    '''
+
+    if isinstance(unit_ids, (int, np.integer)):
+        unit_ids = [unit_ids]
+    elif unit_ids is None:
+        unit_ids = sorting.get_unit_ids()
+    elif not isinstance(unit_ids, (list, np.ndarray)):
+        raise Exception("unit_ids is not a valid in valid")
+    assert np.all([u in sorting.get_unit_ids() for u in unit_ids]), "Invalid unit_ids"
+
+    params_dict = update_all_param_dicts_with_kwargs(kwargs)
+    peak = params_dict['peak']
+    save_property_or_features = params_dict['save_property_or_features']
+    recompute_info = params_dict['recompute_info']
+    assert peak in ['neg', 'pos', 'both'], "'peak' can be 'neg', 'pos', or 'both'"
+
+    if num_channels is None:
+        num_channels = recording.get_num_channels()
+    locations = recording.get_channel_locations()
+
+    coms = []
+    for i, unit_id in enumerate(unit_ids):
+        if 'template' in sorting.get_unit_property_names(unit_id) and not recompute_info:
+            template = sorting.get_unit_property(unit_id, 'template')
+        else:
+            template = get_unit_templates(recording, sorting, unit_id, **kwargs)[0]
+        if peak == 'both':
+            amps = np.max(np.abs(template), 1)
+        elif peak == 'neg':
+            amps = np.min(template, 1)
+        elif peak == 'pos':
+            amps = np.max(template, 1)
+
+
+        idxs = np.argsort(amps)[::-1][:num_channels]
+        com = np.array([np.sum((amps[idxs] * locations[idxs, 0])) / np.sum(amps[idxs]),
+                        np.sum((amps[idxs] * locations[idxs, 1])) / np.sum(amps[idxs])])
```

Comment:
This is OK. I think you can have a direct vector compuation writting without spliting the columns with broacating numpy bility.